### ISSUE TITLE
What is the recommended way to handle random seeds / states in DALI?

### ISSUE THREAD

**Issue by WarBean** [NVIDIA/DALI#1378]
In other data loading approaches like PyTorch 's DataLoader, it is important to properly handle the random seeds / states to (1) ensure reproducibility, (2) successfully resume an interrupted training task from checkpoints. So what is the best practice in DALI? Is there any API for:

- creating a random generator and inform DALI to use it in all augmentation operations
- globally setting random seed for all augmentation process, like `random.seed(1024)`
- set / get the random state for a random generator, like `random.setstate`, `random.getstate`

**Comment by JanuszL** [comment]
Hi,
1) Every operator accepts a `seed` argument (keep in mind that operators with any randomness inside will use them - like readers, support operators, or RandomCrop)
2) You can set seed globally 'seed' in for the whole pipeline and all operators will be initialized based on this one
3) There is no way to obtain a random number generator inside the training. It is rather hard as DALI works asynchronously, so when you obtain the state you will get it probably for one batch ahead then you have consumed.

If you want to mimic something like this you can recreate DALI pipeline each N epochs with the new see and use this as a checkpoint.

**Comment by WarBean** [comment]
@JanuszL Thanks.
